include/lsst/afw/math/TransformBoundedField.h

```diff
@@ -0,0 +1,94 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2017 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_MATH_TransformBoundedField_h_INCLUDED
+#define LSST_AFW_MATH_TransformBoundedField_h_INCLUDED
+
+#include "ndarray.h"
+
+#include "lsst/pex/config.h"
+#include "lsst/afw/geom/Endpoint.h"
+#include "lsst/afw/geom/Transform.h"
+#include "lsst/afw/math/BoundedField.h"
+
+namespace lsst {
+namespace afw {
+namespace math {
+
+/**
+ * A BoundedField based on geom::Transform<Poin2Endpoint, GenericEndpoint<1>>.
+ *
+ * TransformBoundedField supports arbitrary transforms.
+ */
+class TransformBoundedField : public table::io::PersistableFacade<TransformBoundedField>,
+                              public BoundedField {
+public:
+    using Transform = geom::Transform<geom::Point2Endpoint, geom::GenericEndpoint>;
+
+    /**
+     *  Create a TransformBoundedField from a bounding box and transform.
+     */
+    TransformBoundedField(geom::Box2I const& bbox, Transform const& transform);
+
+    /// Get the contained Transform
+    Transform getTransform() const { return _transform; }    
+
+    /// @copydoc BoundedField::evaluate
+    virtual double evaluate(geom::Point2D const & position) const override;
+
+    /// @copydoc BoundedField::evaluate
+    virtual ndarray::Array<double, 1, 1> evaluate(ndarray::Array<double const, 1> const& x,
+                                                  ndarray::Array<double const, 1> const& y) const override;
+
+    using BoundedField::evaluate;
+
+    /// TransformBoundedField is always persistable.
+    virtual bool isPersistable() const override { return true; }
+
+    /// @copydoc BoundedField::operator*
+    virtual std::shared_ptr<BoundedField> operator*(double const scale) const;
+
+    /// @copydoc BoundedField::operator==
+    virtual bool operator==(BoundedField const& rhs) const;
+
+protected:
```

Comment:
I suspect protected is required because these methods aren't called by code in TransformBoundedField, but I'm not sure. I copied ChebyshevBoundedField from @TallJimbo without thinking very hard about it. I propose to leave it for now and consider changing both later, if needed.